service/src/main/java/io/nuvalence/workmanager/service/controllers/TransactionsApiDelegateImpl.java

```
@@ -125,31 +128,24 @@ public ResponseEntity<Map<String, FormConfigurationRenderModel>> getTransactionA
     }
 
     @Override
-    public ResponseEntity<List<TransactionModel>> getTransactionsByUser() {
-        try {
-            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-            if (authentication != null) {
-                if (authentication instanceof UserToken) {
-                    final UserToken token = (UserToken) authentication;
-                    final String userId = token.getApplicationUserId();
+    public ResponseEntity<PagedTransactionModel> getTransactionsForAuthenticatedUser(
+            String sortBy, String sortOrder, Integer page, Integer size) {
 
-                    final List<TransactionModel> transactions =
-                            service.getTransactionsByUser(userId).stream()
-                                    .filter(
-                                            authorizationHandler.getAuthFilter(
-                                                    "view", Transaction.class))
-                                    .map(this::createTransactionModel)
-                                    .collect(Collectors.toList());
+        if (!authorizationHandler.isAllowed("view", Transaction.class)) {
+            throw new ForbiddenException();
+        }
 
-                    return ResponseEntity.ok(transactions);
-                }
-            }
+        final String userId = getAuthenticatedUserId();
 
-            throw new ForbiddenException();
-        } catch (ForbiddenException e) {
-            log.error(AUTH_NOT_SET_ERR_MESSAGE);
-            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        final Page<TransactionModel> transactions =
+                service.getTransactionsByUser(userId, sortBy, sortOrder, page, size)
+                        .map(mapper::transactionToTransactionModel);
+
+        if (transactions.isEmpty()) {
+            throw new NotFoundException();
```

I don't think we want to return an exception in this case as it is likely that a user doesn't have any transactions when they are first starting to use the system

---

ok I just removed that